src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -188,6 +195,101 @@ public override T Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerial
             return default;
         }
 
+        private static bool IsNumericString(ref Utf8JsonReader reader)
+        {
+            ReadOnlySpan<byte> span = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+
+            // Numeric value with Leading and trailing whitespaces should be treated as valid based on behavior of Enum.TryParse()
+#if NETCOREAPP
+            span = span.Trim((byte)' ');
```

Comment:
What about whitespace that isn't the space character?

Reply:
Yes, also need to consider.